sources/packages/backend/apps/workers/src/controllers/assessment/_tests_/e2e/assessment.controller-verifyAssessmentCalculationOrder.e2e-spec.ts

```diff
@@ -269,4 +277,362 @@ describe("AssessmentController(e2e)-verifyAssessmentCalculationOrder", () => {
+            "CSGD",
+            9,
+          ),
+          createFakeDisbursementValue(
+            DisbursementValueType.CanadaGrant,
+            "CSGP",
+            10,
+          ),
+          createFakeDisbursementValue(DisbursementValueType.BCLoan, "BCSL", 11),
+          createFakeDisbursementValue(
+            DisbursementValueType.BCGrant,
+            "SBSD",
+            12,
+          ),
+          createFakeDisbursementValue(
+            DisbursementValueType.BCGrant,
+            "BCAG",
+            13,
+          ),
+        ],
+        programYear: programYear,
+      },
+      {
+        offeringIntensity: OfferingIntensity.fullTime,
+        applicationStatus: ApplicationStatus.Completed,
+        currentAssessmentInitialValues: {
+          assessmentWorkflowId: "some fake id",
+          studentAssessmentStatus: StudentAssessmentStatus.Completed,
+          assessmentDate: addDays(-4),
+        },
+      },
+    );
+    // Current application having the first assessment already processed.
+    const currentApplication = await saveFakeApplicationDisbursements(
+      db.dataSource,
+      { student, programYear: programYear },
+      {
+        offeringIntensity: OfferingIntensity.partTime,
+        applicationStatus: ApplicationStatus.InProgress,
+        currentAssessmentInitialValues: {
+          assessmentWorkflowId: "some fake id",
+          studentAssessmentStatus: StudentAssessmentStatus.Completed,
+          assessmentDate: addDays(-3),
+        },
+      },
+    );
+    const firstAssessmentDate =
+      currentApplication.currentAssessment.assessmentDate;
+    // Create the second assessment for the current application with a different assessment date.
+    const secondAssessment = createFakeStudentAssessment(
+      {
+        auditUser: currentApplication.student.user,
+        application: currentApplication,
+        offering: currentApplication.currentAssessment.offering,
+      },
+      {
+        initialValue: {
+          ...currentApplication.currentAssessment,
+          assessmentWorkflowId: "some fake id",
+          studentAssessmentStatus: StudentAssessmentStatus.InProgress,
+          assessmentDate: addDays(-1),
+        },
+      },
+    );
+    currentApplication.currentAssessment = secondAssessment;
+    await db.dataSource.getRepository(Application).save(currentApplication);
+
+    // SFAS Individual.
+    const sfasIndividual = await saveFakeSFASIndividual(db.dataSource, {
+      initialValues: {
+        lastName: student.user.lastName,
+        birthDate: student.birthDate,
+        sin: student.sinValidation.sin,
+        unsuccessfulCompletion: 12,
+      },
+    });
+    // SFAS application with the end date as the end date of the program year.
+    await saveFakeSFASApplication(db.dataSource, {
+      initialValues: {
+        startDate: currentApplication.programYear.startDate,
+        endDate: currentApplication.programYear.endDate,
+        programYearId: 20232024,
```

Comment:
Program year id is not required to be set as we evaluate based on start and end date.